### ISSUE TITLE
Bug: RunIf doesn't work in a bash task

### ISSUE THREAD

**Issue by asavchkov** [gocd-contrib/gocd-groovy-dsl-config-plugin#363]
This part of a pipeline:
```groovy
                tasks {
                  bash {commandString = 'false'}
                  bash {
                    commandString = 'echo I am bash with runIf=failed'
                    runIf = 'failed'
                  }
                  exec {
                    commandLine = ['echo', 'I am exec with runIf=failed']
                    runIf = 'failed'
                  }
                  bash {
                    commandString = 'echo I am bash with runIf=any'
                    runIf = 'any'
                  }
                  exec {
                    commandLine = ['echo', 'I am exec with runIf=any']
                    runIf = 'any'
                  }
                }
```
produces this result:
```
##|01:55:12.644 [go] Start to build My-pipeline/17/Build/1/Test-runIf on my-agent [/var/lib/go-agent]
!!|01:55:12.655 [go] Task: bash -c false
?1|01:55:12.766 [go] Task status: failed (110 ms) (exit code: 1)
!!|01:55:12.766 [go] Task: echo "I am exec with runIf=failed"
&1|01:55:12.771 I am exec with runIf=failed
?0|01:55:12.870 [go] Task status: passed (103 ms)
!!|01:55:12.871 [go] Task: echo "I am exec with runIf=any"
&1|01:55:12.874 I am exec with runIf=any
?0|01:55:12.974 [go] Task status: passed (103 ms)
j1|01:55:12.991 [go] Current job status: failed
```
that is the `bash` tasks with `runIf` are not executed. According to the javadoc both the `bash` and the `exec` tasks have a `runIf` property so I would expect that both of them should be executed in the same way in the example above.

DSL plugin version: 2.1.1-432
GoCD server version: 22.3.0 (15301-9d23ed19a9ea46eaf7f18bd16671ae0569871f53)
GoCD agent version: go-agent_22.3.0-15301_all.deb

**Comment by chadlwilson** [comment]
Thanks for reporting! Should be fixed in [`2.1.2`](https://github.com/gocd-contrib/gocd-groovy-dsl-config-plugin/releases/tag/v2.1.2-446) - let me know if you still have a problem.

**Comment by asavchkov** [comment]
It is working as expected now, many thanks for fixing the bug so quickly.
